crates/flux-refineck/src/checker.rs

```diff
@@ -703,6 +706,39 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
         self.check_closure_clauses(infcx, &obligations, span)
     }
 
+    fn unfold_local_ptrs(
+        infcx: &mut InferCtxt<'_, 'genv, 'tcx>,
+        env: &mut TypeEnv,
+        span: Span,
+        fn_sig: &EarlyBinder<PolyFnSig>,
+        actuals: &[Ty],
+    ) -> Result<Vec<Ty>> {
+        // We *only* need to know whether each input is a &strg or not
+        let mut at = infcx.at(span);
```

Comment:
I believe unfolding doesn't generate any constraints so we don't need the `infcx.at` (that's a nice static guarantee if we don't create an `InferCtxtAt` then we know we didn't generate any constraints).

```suggestion
```